packages/getters/src/swap-view.ts

```diff
@@ -25,56 +29,48 @@ export const getOutput2Value = createGetter((swapView?: SwapView) => {
   }
 });
 
-// Generic getter function for 'SwapPlaintext' and 'SwapBody' within SwapView
-// const createSwapGetter = <T, K extends keyof T>(property: K) => {
-//   return createGetter((swapView?: SwapView) => {
-//     switch (swapView?.swapView.case) {
-//       case 'visible':
-//         return swapView.swapView.value.swapPlaintext?.[property as keyof SwapPlaintext] as T[K];
-//       case 'opaque':
-//         return swapView.swapView.value.swap?.body?.[property as keyof SwapBody] as T[K];
-//       default:
-//         return undefined;
-//     }
-//   });
-// };
+// Generic getter function that returns the value of a specified property from either 'swapPlaintext' or 'swapBody'.
+// This pattern utilizes parameterized types, 'T', to handle property access, 'K' within different nested objects based on the case
+// of 'SwapView'. These parameterized types can represent an intersection of multiple types.
+const createSwapGetter = <T, K extends keyof T>(property: K) => {
```

Comment:
I modified to getter to ensure an `Amount` is returned to the caller. I quite like this pattern, and maybe it's extensible to other getter functions in the future.

Reply:
Unfortunately, this re-introduces the bug 🐛 with `claimFee` in the opaque case.

Using `as T[K]` is basically saying that we guarantee every field of `SwapPlaintext` and `SwapBody` are the same. But it then allows for human error. This helper then only becomes useful if the consumer will only use it in the correct way.

In short, the abstraction is not worth it. Think we should revert to previous version.

Reply:
would it? wouldn't the opaque case just return undefined because there isn't a `claimFee` field?

Reply:
that makes sense! yeah, the parameterization `T` represents an _intersection_ rather than a union. I think keeping this pattern is a good reference for eventually overhauling other getters to be generic over the `visible` and `opaque` views down the line 👍

Reply:
I'll open an issue to add more detailed comments to our getters package